crates/circuit/src/bit.rs

```diff
@@ -859,6 +859,13 @@ impl ShareableQubit {
     pub fn is_ancilla(&self) -> bool {
         *self.subclass() == QubitSubclass::ANCILLA
     }
+
+    pub fn index(&self) -> usize {
+        match &self.0 {
+            BitInfo::Owned { index, .. } => *index as usize,
+            BitInfo::Anonymous { .. } => panic!("Anonymous qubit has no index"),
+        }
+    }
```

Comment:
Having this panic may be a bit extreme, remember that a panic is an unrecoverable error. Perhaps having this method return an `Option<usize>` might be better since the user would be able to easily handle a case where an index doesn't exist.